packages/2d/src/Components/Geometry.ts

```diff
@@ -13,16 +13,21 @@ import {
  *
  * You should only have one `Geometry` component per `Entity`.
  */
-function Geometry<S extends Shape>({
```

Comment:
Oh, I still need this generic, though. Otherwise you can't use Circle-specific or Polygon-specific methods on the Geometry's shape after it's initialized.